purchases/src/main/kotlin/com/revenuecat/purchases/PurchaseParams.kt

```diff
@@ -161,22 +169,48 @@ class PurchaseParams(val builder: Builder) {
          */
         @ExperimentalPreviewRevenueCatPurchasesAPI
         fun addOnStoreProducts(addOnStoreProducts: List<StoreProduct>) = apply {
-            if (addOnStoreProducts.isEmpty()) {
+            val compatibleAddOnProducts: List<GooglePurchasingData> = addOnStoreProducts
+                .mapNotNull { it.purchasingData as? GooglePurchasingData.Subscription }
+
+            attachSubscriptionAddOns(addOns = compatibleAddOnProducts)
+        }
+
+        /*
+         * The [SubscriptionOption]s to add on to the base product passed in via
+         * the [PurchaseParams.Builder]'s constructor. This will result in a multi-line purchase whose base product
+         * is the one passed in to the [PurchaseParams.Builder]'s constructor.
+         *
+         * The following restrictions apply to add-on purchases:
+         * - Add-on purchases are currently only supported for subscriptions on the Play Store.
+         * - The renewal periods of all add-on products must be the same and match the period of the base item.
+         * - No more than 49 add-ons products per multi-line purchase are allowed.
+         */
+        @ExperimentalPreviewRevenueCatPurchasesAPI
+        fun addOnSubscriptionOptions(addOnSubscriptionOptions: List<SubscriptionOption>) = apply {
+            val compatibleAddOnProducts: List<GooglePurchasingData> = addOnSubscriptionOptions
+                .mapNotNull { it.purchasingData as? GooglePurchasingData.Subscription }
+
+            attachSubscriptionAddOns(addOns = compatibleAddOnProducts)
+        }
+
+        @OptIn(ExperimentalPreviewRevenueCatPurchasesAPI::class)
+        private fun attachSubscriptionAddOns(addOns: List<GooglePurchasingData>) = apply {
+            if (addOns.isEmpty()) {
```

Comment:
NABD but should we just return after logging in this case?

Reply:
Makes sense! I've added that return here: ae6763d09a294f3049ec004a7025e83056fe83ba